packages/server-wallet/src/protocols/actions.ts

```diff
@@ -9,7 +9,7 @@ Actions that protocols can declare.
 
 type UpdateOpts = Partial<StateVariables>;
 type UpdateChannel = {type: 'UpdateChannel'; channelId: Bytes32} & UpdateOpts;
-type SignState = {type: 'SignState'; channelId: Bytes32; hash: Bytes32};
+export type SignState = {type: 'SignState'; channelId: Bytes32} & UpdateOpts;
```

Comment:
I'm assuming that `SignState` will
- Sign the state
- Update the store with the state
- Communicate the state to opponents.

I'm not sure we need an `UpdateChannel` and `SignState`?

Reply:
It might be a worthwhile separation:
- `UpdateChannel` will create a _new_ state, and sign it.
- `SignState` will sign an _existing_ state

Reply:
> It might be a worthwhile separation:
> 
> * `UpdateChannel` will create a _new_ state, and sign it.
> * `SignState` will sign an _existing_ state

How would `SignState` specify the existing state? I don't think just signing whatever the latest state happens to be would be safe.

If we plan to specify them by using `Partial<StateVariables>` then I'm not sure if the distinction between new or existing states matter?  We just want a signed state with a certain turnnumber, outcome, etc.. It doesn't matter if we're creating it from scratch or supporting an existing state (I think?)

Reply:
> > It might be a worthwhile separation:
> > 
> > * `UpdateChannel` will create a _new_ state, and sign it.
> > * `SignState` will sign an _existing_ state
> 
> How would `SignState` specify the existing state? I don't think just signing whatever the latest state happens to be would be safe.

The current `Channel` model implements [this](https://github.com/statechannels/statechannels/blob/master/packages/server-wallet/src/models/channel.ts#L155), which is a change from the `ChannelStoreEntry`.

> If we plan to specify them by using `Partial<StateVariables>` then I'm not sure if the distinction between new or existing states matter? We just want a signed state with a certain turnnumber, outcome, etc.. It doesn't matter if we're creating it from scratch or supporting an existing state (I think?)

I'm actually thinking it's beneficial to specify _all_ `StateVariables` in the `UpdateChannel` action, to help deal with race conditions. In that case, there's no difference between `SignState` and `UpdateChannel` -- they can both be `SignState`, with some specific `StateVariables`.